cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -195,11 +193,11 @@ public void dropAllInNamespace(final NamespaceId namespace) throws Exception {
     });
 
     for (final Location location : locations) {
-      doDrop(namespace.stream(StreamUtils.getStreamNameFromLocation(location)), location, ugi);
+      doDrop(namespace.stream(StreamUtils.getStreamNameFromLocation(location)), location);
     }
 
     // Also drop the state table
-    ImpersonationUtils.doAs(ugi, new Callable<Void>() {
+    impersonator.doAs(namespace, new Callable<Void>() {
```

Comment:
I think admin being able to delete streams is the correct behavior. They should not be able to see or modify the data without explicit read and write permissions but creating and deleting are admin operations. Whether we will be able to delete will need to be tested. If this does not work then we will need to find a way to delete each table as the table owner.